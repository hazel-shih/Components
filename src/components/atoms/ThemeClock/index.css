:root {
	--second-needle-degree: var(--second-needle-degree);
	--minute-needle-degree: var(--minute-needle-degree);
	--hour-needle-degree: var(--hour-needle-degree);
}

.theme-clock {
	width: fit-content;
	& > .clock {
		width: 200px;
		height: 200px;
		border: 2px solid var(--neutral-n1-primary-text);
		position: relative;
		border-radius: 50%;

		& > .needle {
			position: absolute;
			height: 65px;
			width: 3px;
			background-color: var(--neutral-n1-primary-text);
			left: 50%;
			top: 50%;
			transform-origin: bottom center;

			&.hour {
				width: 7px;
				height: 55px;
				transform: translate(-50%, -100%) rotate(var(--hour-needle-degree));
			}

			&.minute {
				width: 5px;
				height: 70px;
				transform: translate(-50%, -100%) rotate(var(--minute-needle-degree));
			}

			&.second {
				height: 70px;
				transform: translate(-50%, -100%) rotate(var(--second-needle-degree));
				background-color: var(--secondary-pressed);
			}
		}

		& > .center-point {
			position: absolute;
			top: 50%;
			left: 50%;
			transform: translate(-50%, -50%);
			width: 10px;
			height: 10px;
			background-color: var(--red-r3-alert);
			border-radius: 50%;
		}
	}
	& > .number-time {
		margin-top: 30px;
		display: flex;
		flex-direction: column;
		justify-content: center;
		align-items: center;
		font-size: 18px;
		color: var(--neutral-n1-primary-text);
	}
}
