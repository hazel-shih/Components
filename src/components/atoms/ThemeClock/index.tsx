import React, { useEffect, useState, CSSProperties } from 'react';
import classnames from 'classnames';
import { getHours, getMinutes, getSeconds, format } from 'date-fns';

import styles from './index.css';

interface ThemeClockProperty {
	className?: string;
}

const ThemeClock: React.FC<ThemeClockProperty> = ({ className }) => {
	const [currentTime, setCurrentTime] = useState<Date>(new Date());
	useEffect(() => {
		setTimeout(() => {
			setCurrentTime(new Date());
		}, 1000);
	}, [currentTime]);

	return (
		<div className={classnames(styles.themeClock, className)}>
			<div className={styles.clock}>
				<div
					className={classnames(styles.needle, styles.hour)}
					style={{ '--hour-needle-degree': `${getHours(currentTime) * 6}deg` } as CSSProperties}
				/>
				<div
					className={classnames(styles.needle, styles.minute)}
					style={{ '--minute-needle-degree': `${getMinutes(currentTime) * 6}deg` } as CSSProperties}
				/>
				<div
					className={classnames(styles.needle, styles.second)}
					style={{ '--second-needle-degree': `${getSeconds(currentTime) * 6}deg` } as CSSProperties}
				/>
				<div className={styles.centerPoint} />
			</div>
			<div className={styles.numberTime}>
				<p>{format(new Date(currentTime), 'yyyy / MM / dd')}</p>
				<p>{format(new Date(currentTime), 'hhï¼šmm')}</p>
			</div>
		</div>
	);
};

export default ThemeClock;
