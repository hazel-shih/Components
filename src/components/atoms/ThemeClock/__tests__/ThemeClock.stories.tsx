import React from 'react';
import { storiesOf } from '@storybook/react';

import ThemeClock from 'components/atoms/ThemeClock';

const stories = storiesOf('atoms/ThemeClock', module);

stories.add('__interactive', () => <ThemeClock />);
