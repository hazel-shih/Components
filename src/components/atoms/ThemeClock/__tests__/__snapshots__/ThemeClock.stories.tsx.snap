// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <div
    class="themeClock"
  >
    <div
      class="clock"
    >
      <div
        class="needle hour"
        style="--hour-needle-degree: 48deg;"
      />
      <div
        class="needle minute"
        style="--minute-needle-degree: 0deg;"
      />
      <div
        class="needle second"
        style="--second-needle-degree: 0deg;"
      />
      <div
        class="centerPoint"
      />
    </div>
    <div
      class="numberTime"
    >
      <p>
        2019 / 07 / 01
      </p>
      <p>
        12ï¼š00
      </p>
    </div>
  </div>
</DocumentFragment>
`;
